src/test/java/au/org/ala/names/util/TestUtils.java

```diff
@@ -0,0 +1,63 @@
+package au.org.ala.names.util;
+
+import au.org.ala.names.index.NameProvider;
+import au.org.ala.names.index.TaxonConceptInstance;
+import au.org.ala.names.model.RankType;
+import au.org.ala.names.model.TaxonomicType;
+import org.gbif.api.vocabulary.NomenclaturalCode;
+import org.gbif.api.vocabulary.Rank;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+
+/**
+ * Handy test utilities.
+ *
+ * @author Doug Palmer &lt;Doug.Palmer@csiro.au&gt;
+ * @copyright Copyright &copy; 2017 Atlas of Living Australia
+ */
+public class TestUtils {
+    /**
+     * Load a class resource into a string.
+     * <p>
+     * Useful for comparing generated data against a file.
+     * </p>
+     * @param path The path relative to the class
+     *
+     * @return The resource as a string
+     *
+     * @throws IOException If unable to read the resource
+     */
+    public String loadResource(String path) throws IOException {
+        Reader reader = this.resourceReader(path);
+        StringBuffer sb = new StringBuffer(1024);
```

Comment:
Not a big issue, but ``StringBuilder`` is the unsynchronized version of ``StringBuffer``.

Reply:
You could also change StringBuffer to StringWriter and use IOUtils.copy:

https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/IOUtils.html#copy(java.io.Reader,%20java.io.Writer)

Reply:
Noice